daemon/container.go

```diff
@@ -262,9 +264,16 @@ func (container *Container) Start() (err error) {
 	defer func() {
 		if err != nil {
 			container.setError(err)
+			if strings.Contains(err.Error(), syscall.ENOENT.Error()) ||
```

Comment:
Added comment https://github.com/docker/docker/pull/14012/files#diff-07773483549b2d3c003c53614e1e2ab2R45 to clarify.